src/lib.rs

```diff
@@ -511,8 +515,18 @@ impl FrankaSim {
         0.02,
     ];
     const PANDA_NUM_DOFS: usize = 7;
-    const PANDA_END_EFFECTOR_INDEX: usize = 11;
     const GRIPPER_OPEN_WIDTH: f64 = 0.08;
+    const ROBOT_JOINT_NAMES: [&'static str; 7] = [
+        "panda_joint1",
+        "panda_joint2",
+        "panda_joint3",
+        "panda_joint4",
+        "panda_joint5",
+        "panda_joint6",
+        "panda_joint7",
+    ];
+    const GRIPPER_JOINT_NAMES: [&'static str; 2] = ["panda_finger_joint1", "panda_finger_joint2"];
```

Comment:
Could introduce and use `FrankaSim::FRANKA_HAND_DOFS` instead of magic number.

Generally you have a lot of magic numbers like `7`, `2` and even `9` (7+2) in your code base. Why not replace them with properly named constants?

Reply:
As the 2 is only used in two other places, so I settled for GRIPPER_JOINT_NAMES.len(). For the 9 I did

```rust
 const PANDA_NUM_TOTAL_DOFS: usize =
        FrankaSim::PANDA_NUM_DOFS + FrankaSim::GRIPPER_JOINT_NAMES.len();
```